src/GovUk.Education.ExploreEducationStatistics.Content.Services/RedirectsService.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services;
+
+public class RedirectsService : IRedirectsService
+{
+    private readonly ContentDbContext _contentDbContext;
+
+    public RedirectsService(ContentDbContext contentDbContext)
+    {
+        _contentDbContext = contentDbContext;
+    }
+
+    public async Task<Either<ActionResult, RedirectsViewModel>> List()
+    {
+        // A redirect for a methodologyVersion that isn't published shouldn't appear in the list. A redirect becomes
+        // active once the associated methodologyVersion is published. It remains active if subsequent methodology
+        // amendment's are published. We establish this by checking against each methodologyVersion's Version.
+        var methodologyRedirects = await _contentDbContext.MethodologyRedirects
+            .Where(mr => mr.MethodologyVersion.Methodology.LatestPublishedVersion != null
+                         && mr.MethodologyVersion.Methodology.LatestPublishedVersion.Version >=
+                         mr.MethodologyVersion.Version
+            )
+            .Select(mr => new
+            {
+                RedirectSlug = mr.Slug,
+                // MethodologyVersion.Slug cannot be translated into SQL, so we do this...
+                LatestPublishedSlug = mr.MethodologyVersion.Methodology.LatestPublishedVersion!.AlternativeSlug
+                                      ?? mr.MethodologyVersion.Methodology.OwningPublicationSlug,
+            })
+            .ToListAsync();
+
+        var methodologyRedirectViewModels = methodologyRedirects
+            .Select(mr =>
+            {
+                if (mr.RedirectSlug == mr.LatestPublishedSlug)
+                {
+                    // It is possible for a redirect to point to the currently active slug. This can happen as a
+                    // methodology can change to a slug even if there is a redirect for that slug,
+                    // *if* the redirect is for the same methodology - i.e. the methodology's slug is changing
+                    // back to a previous slug it used. But the slug may not change back immediately, if the change
+                    // is on an unpublished amendment, so we cannot remove the redirect immediately when changing the
+                    // slug back; redirect needs to be active until the amendment is published.
+                    return null;
```

Comment:
Should we instead be maintaining a "Published" DateTime on MethodologyRedirects itself that is set and unset by the Publisher to make it clearer which redirects are active at any given time, or are you happy with the way that we have it here?

Reply:
I think I'm happy without it. My approach was to attempt to avoid having to make changes during publishing wherever possible. So the logic for whether a methodology redirect is active lives solely in `RedirectsService#List`. My general thought was that makes the feature more contained, as you just have to create the redirects, not worry about when they should be published. If the publisher is refactored, you only need to worry about updating the cache.

I'm inclined to think that was the simpler approach, but honestly not sure if that is right. One of those instances where it would be nice to see both implemented to more easily compare them.

Not sure how much work it would be to switch. I don't think it would be a lot of code - most of the time would probably be spent making sure I'd not broken anything. If you feel it would be worth it, I'm happy to give it a go.